composer/utils/checkpoint.py

```diff
@@ -68,6 +68,23 @@ def _get_write_mode(name: str) -> str:
     raise ValueError(f'{name} does not end with a valid tarfile extension.')
 
 
+class PartialFilePath:
```

Comment:
Is this a suffix of a filepath? If so, can we name accordingly?

Reply:
It's not just a suffix, but also handles the formatting (hence the word `Partial`). Does that make sense, or need a different name?